File: apps/app-frontend/src/helpers/onboarding-checklist.ts
import { invoke } from '@tauri-apps/api/core'

import type { OnboardingChecklist } from '@/generated/app-events/OnboardingChecklist'

export type { OnboardingChecklist }

export async function getOnboardingChecklist(): Promise<OnboardingChecklist> {
	return await invoke('plugin:onboarding-checklist|get_onboarding_checklist')
}

/** Complete the site-account onboarding step after Owyx email login. */
export async function markLoggedIntoOwyxSite(): Promise<void> {
	await invoke('plugin:onboarding-checklist|mark_logged_into_owyx_site')
}

File: apps/app/src/api/onboarding_checklist.rs
use crate::api::Result;
use tauri::plugin::TauriPlugin;
use theseus::prelude::*;

pub fn init<R: tauri::Runtime>() -> TauriPlugin<R> {
    tauri::plugin::Builder::new("onboarding-checklist")
        .invoke_handler(tauri::generate_handler![
            get_onboarding_checklist,
            mark_logged_into_owyx_site
        ])
        .build()
}

#[tauri::command]
pub async fn get_onboarding_checklist() -> Result<OnboardingChecklist> {
    Ok(onboarding_checklist::get().await?)
}

/// Marks the site-account onboarding step (same DB flag as legacy Modrinth login).
#[tauri::command]
pub async fn mark_logged_into_owyx_site() -> Result<()> {
    Ok(onboarding_checklist::mark_logged_into_modrinth().await?)
}
